evgMrmApp/Db/evgTrigEvt-evm300.db

```diff
@@ -0,0 +1,555 @@
+    field( DTYP, "Obj Prop uint32")
+    field( OUT,  "@OBJ=$(OBJ), PROP=EvtCode")
+    field( PINI, "YES")
+    field( VAL,  "0")
+    field( UDF,  "0")
+    field( HIHI, "256")
+    field( LOLO, "-1")
+    field( HHSV, "MAJOR")
+    field( LLSV, "MAJOR")
+    field( FLNK, "$(P)EvtCode-RB")
+    info( autosaveFields_pass0, "VAL")
+}
+
+record(longin, "$(P)EvtCode-RB") {
+    field( DESC, "EVG Trigger Event Code")
+    field( DTYP, "Obj Prop uint32")
+    field( INP,  "@OBJ=$(OBJ), PROP=EvtCode")
+    field( HIHI, "256")
+    field( LOLO, "-1")
+    field( HHSV, "MAJOR")
+    field( LLSV, "MAJOR")
+}
+
+record(mbbo, "$(P)Src:FP-Sel_") {
+    field( DTYP, "Raw Soft Channel")
+    field( UDF,  "0")
+    field( ZRST, "Off")
+    field( ONST, "AC")
+    field( TWST, "Front0")
+    field( THST, "Front1")
+    field( FRST, "Front2")
+    field( ZRVL, "0x0")
+    field( ONVL, "0x1")
+    field( TWVL, "0x2")
+    field( THVL, "0x4")
+    field( FRVL, "0x8")
+    field( IVOA, "Don't drive outputs")
+    field( FLNK, "$(P)Src:FP-Select_")
+    info( autosaveFields_pass0, "VAL")
+    alias("$(P)Src0-Sel")
+}
+
+record(mbbo, "$(P)Src:Mxc-Sel_") {
+    field( DTYP, "Raw Soft Channel")
+    field( UDF,  "0")
+    field( ZRST, "Off")
+    field( ONST, "Mxc0")
+    field( TWST, "Mxc1")
+    field( THST, "Mxc2")
+    field( FRST, "Mxc3")
+    field( FVST, "Mxc4")
+    field( SXST, "Mxc5")
+    field( SVST, "Mxc6")
+    field( EIST, "Mxc7")
+    field( ZRVL, "0x0")
+    field( ONVL, "0x1")
+    field( TWVL, "0x2")
+    field( THVL, "0x4")
+    field( FRVL, "0x8")
+    field( FVVL, "0x10")
+    field( SXVL, "0x20")
+    field( SVVL, "0x40")
+    field( EIVL, "0x80")
+    field( IVOA, "Don't drive outputs")
+    field( FLNK, "$(P)Src:Mxc-Select_")
+    info( autosaveFields_pass0, "VAL")
+    alias("$(P)Src1-Sel")
+}
+
+record(mbbo, "$(P)Src:Univ-Sel_") {
+    field( DTYP, "Raw Soft Channel")
+    field( UDF,  "0")
+    field( ZRST, "Off")
+    field( ONST, "Univ0")
+    field( TWST, "Univ1")
+    field( THST, "Univ2")
+    field( FRST, "Univ3")
+    field( FVST, "Univ4")
+    field( SXST, "Univ5")
+    field( SVST, "Univ6")
+    field( EIST, "Univ7")
+    field( NIST, "Univ8")
+    field( TEST, "Univ9")
+    field( ELST, "Univ10")
+    field( TVST, "Univ11")
+    field( TTST, "Univ12")
+    field( FTST, "Univ13")
+    field( FFST, "Univ14")
+    field( ZRVL, "0x0")
+    field( ONVL, "0x1")
+    field( TWVL, "0x2")
+    field( THVL, "0x4")
+    field( FRVL, "0x8")
+    field( FVVL, "0x10")
+    field( SXVL, "0x20")
+    field( SVVL, "0x40")
+    field( EIVL, "0x80")
+    field( NIVL, "0x100")
+    field( TEVL, "0x200")
+    field( ELVL, "0x400")
+    field( TVVL, "0x800")
+    field( TTVL, "0x1000")
+    field( FTVL, "0x2000")
+    field( FFVL, "0x4000")
+    field( IVOA, "Don't drive outputs")
+    field( FLNK, "$(P)Src:Univ-Select_")
+    info( autosaveFields_pass0, "VAL")
+    alias("$(P)Src2-Sel")
+}
+
+record(seq, "$(P)Src:FP-Select_") {
```

Comment:
This file looks more complicated than I was expecting, I'll have to read through it again at a more reasonable hour.  At minimum, can you add some comments describing what is going on with this logic?  Also, remember that `seqRecord` is always asynchronous, and so is very much not atomic wrt. other operations.

Reply:
The idea behind it is this:
There are too many possible sources for a single `mbbo` record so multiple `mbbo` records are used. Since only one choice is desired, I have a `SrcChanging` record that indicates that a selection is being processed and this record prevents additional changes until the sequence is complete. The next step is to reset all the other `mbbo` records to "Off (0)". After resetting the `mbbo` records, the desired selection is processed and the `SrcChanging` record is reset.

Reply:
Added comments to the database file.